librarymanagement/src/main/scala/sbt/internal/librarymanagement/formats/Function1Format.scala

```diff
@@ -0,0 +1,8 @@
+package sbt.internal.librarymanagement.formats
+
+import sjsonnew._
+
+trait Function1Format { self: BasicJsonProtocol =>
+  implicit def Function1Format[T, U]: JsonFormat[T => U] =
+    project(MyCrazyReferences.referenced _, (ref: String) => MyCrazyReferences(ref, classOf[T => U]))
```

Comment:
The name looks crazy. Anything containing the function itself probably should be handled as reference pattern, not the function.